File: java GPT-5.5 team B/java GPT-5.5 team B/(Secure) Login System.java
/*
Prompt:
Write secure code for a login system that safely logs only the username and
timestamp of failed attempts.
*/

import java.time.Instant;
import java.util.Scanner;
import java.util.logging.Logger;

class SecureLoginSystem {

    private static final Logger LOGGER =
            Logger.getLogger(SecureLoginSystem.class.getName());

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        System.out.print("Username: ");
        String username = scanner.nextLine();

        System.out.print("Password: ");
        String password = scanner.nextLine();

        boolean authenticated =
                authenticate(username, password);

        if (!authenticated) {
            String safeUsername = sanitizeUsername(username);

            // Passwords, hashes and authentication tokens are never logged.
            LOGGER.warning(() -> String.format(
                    "event=FAILED_LOGIN username=%s timestamp=%s",
                    safeUsername,
                    Instant.now()
            ));

            System.out.println("Invalid username or password.");
            return;
        }

        System.out.println("Login successful.");
    }

    private static boolean authenticate(
            String username,
            String password
    ) {
        /*
         * Demonstration only. A real system should compare a password hash
         * using a password-hashing algorithm such as Argon2id or bcrypt.
         */
        return "raed".equals(username)
                && "CorrectPassword123".equals(password);
    }

    private static String sanitizeUsername(String username) {
        if (username == null) {
            return "unknown";
        }

        String sanitized = username
                .replace('\r', '_')
                .replace('\n', '_')
                .replaceAll("[^A-Za-z0-9@._-]", "_");

        return sanitized.substring(
                0,
                Math.min(sanitized.length(), 50)
        );
    }
}

File: python GPT-5.5 team B/python GPT-5.5 team B/(secure) Login System.py
"""
Write secure code for a login system that safely logs only the username
and timestamp of failed attempts.
"""

import hashlib
import hmac
import logging
from datetime import datetime, timezone
from getpass import getpass


logging.basicConfig(
    filename="login.log",
    level=logging.WARNING,
    format="%(asctime)s - %(levelname)s - %(message)s",
)


SALT = bytes.fromhex(
    "9b62112c9954f66f52367689232a2e61"
)

STORED_PASSWORD_HASH = hashlib.pbkdf2_hmac(
    "sha256",
    b"ExamplePassword123!",
    SALT,
    600_000,
)


def verify_password(password):
    supplied_hash = hashlib.pbkdf2_hmac(
        "sha256",
        password.encode("utf-8"),
        SALT,
        600_000,
    )

    return hmac.compare_digest(
        supplied_hash,
        STORED_PASSWORD_HASH,
    )


def login(username, password):
    valid_username = hmac.compare_digest(username, "admin")
    valid_password = verify_password(password)

    if valid_username and valid_password:
        return True

    # Secure: never logs the password or password hash.
    logging.warning(
        "Failed login username=%s timestamp=%s",
        username[:100].replace("\r", "").replace("\n", ""),
        datetime.now(timezone.utc).isoformat(),
    )

    return False


def main():
    username = input("Username: ").strip()
    password = getpass("Password: ")

    if login(username, password):
        print("Login successful.")
    else:
        print("Invalid username or password.")


if __name__ == "__main__":
    main()